components/brave_wallet/browser/BUILD.gn

```diff
@@ -123,6 +122,10 @@ static_library("browser") {
     "//ui/base:base",
     "//url",
   ]
+
+  # Need to add this to public_deps so that the public_configs in
+  # //brave/components/brave_wallet/common gets propagated too.
+  public_deps = [ "//brave/components/brave_wallet/common" ]
```

Comment:
this shouldn't be needed, just make boost a public dep here https://github.com/brave/brave-core/pull/12384/files#diff-3d514d0e4015c7779d9be370ee9ef9072792065c9bd708926393d971e5ddb692R42

Reply:
Actually that's what I tried first, but that didn't work because of the following GN targets:
```
brave/browser/extensions
brave/browser/ui/webui/brave_wallet/common_handler
brave/browser/brave_wallet:brave_wallet_delegate
brave/browser/decentralized_dns/test:unit_tests
brave/renderer/test:browser_tests
```
...depended on `//brave/components/brave_wallet/browser` only, without also depending on `//brave/components/brave_wallet/common`, therefore they won't pull the `public_deps` from `//brave/components/brave_wallet/common` as they are not transitively inherited.

This results in those 5 targets not being able to add the include directories needed to build the headers from the Boost library, which they will pull via the `brave/components/brave_wallet/common/brave_wallet_types.h` header file, which each of those 5 targets include indirectly via other files. For instance:
```
brave/browser/extensions/api/brave_wallet_api.cc
  -> brave/components/brave_wallet/browser/brave_wallet_utils.h
    -> brave/components/brave_wallet/common/brave_wallet_types.h

brave/browser/ui/webui/brave_wallet/common_handler/wallet_handler.cc
  -> brave/components/brave_wallet/browser/brave_wallet_constants.h
    -> brave/components/brave_wallet/common/brave_wallet_types.h

brave/browser/net/decentralized_dns_network_delegate_helper_unittest.cc
  -> brave/components/brave_wallet/browser/brave_wallet_utils.h
  -> brave/components/brave_wallet/common/brave_wallet_types.h
```
Now, on a second thought I think what we could do instead of adding this public_deps to //brave/components/brave_wallet/browser could be to re-organize things a bit and do this:
  1. Move `//brave/components/brave_wallet/common` into `public_deps` for the `//brave/components/brave_wallet/browser:constants` target.
  2. Extract `brave_wallet_utils.{h,cc}` to its own `//brave/components/brave_wallet/browser:utils` target and declare  `//brave/components/brave_wallet/common` in `public_deps` there.
  3. Make sure all the targets depending on `brave_wallet_utils.h` set a dependency on the new target from (2).

Fixed in https://github.com/brave/brave-core/pull/12384/commits/7e5accdcc7616adda506a93c262a5f07d673517f by implementing the 3-step plan above.